datafusion/core/src/datasource/listing/url.rs

```diff
@@ -87,6 +89,41 @@ impl ListingTableUrl {
         }
     }
 
+    /// Perform shell-like path expansions
+    /// * Home directory expansion: "~/test.csv" expands to "/Users/user1/test.csv"
+    /// * Environment variable expansion: "$HOME/$DATA/test.csv" expands to
+    /// "/Users/user1/data/test.csv"
+    fn expand_path_prefix(path: &str) -> Result<String, DataFusionError> {
+        let home_dir = home::home_dir()
+            .unwrap()
```

Comment:
I don't think we should use either `unwrap`  as it will cause a panic of the user doesn't have a $HOME set, I think. Instead I think the code should simply not attempt `~` substitution if it can't find the home directory expansion